modules/core/src/v2/coinFactory.ts

```diff
@@ -68,137 +68,228 @@ import { Tbcha } from './coins/tbcha';
-GlobalCoinFactory.registerCoinConstructor('ltc', Ltc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tltc', Tltc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('eos', Eos.createInstance);
-GlobalCoinFactory.registerCoinConstructor('teos', Teos.createInstance);
-GlobalCoinFactory.registerCoinConstructor('eth', Eth.createInstance);
-GlobalCoinFactory.registerCoinConstructor('teth', Teth.createInstance);
-GlobalCoinFactory.registerCoinConstructor('gteth', Gteth.createInstance);
-GlobalCoinFactory.registerCoinConstructor('eth2', Eth2.createInstance);
-GlobalCoinFactory.registerCoinConstructor('teth2', Teth2.createInstance);
-GlobalCoinFactory.registerCoinConstructor('etc', Etc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tetc', Tetc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('rbtc', Rbtc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('trbtc', Trbtc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('celo', Celo.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tcelo', Tcelo.createInstance);
-GlobalCoinFactory.registerCoinConstructor('avaxc', AvaxC.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tavaxc', TavaxC.createInstance);
-GlobalCoinFactory.registerCoinConstructor('xrp', Xrp.createInstance);
-GlobalCoinFactory.registerCoinConstructor('txrp', Txrp.createInstance);
-GlobalCoinFactory.registerCoinConstructor('xlm', Xlm.createInstance);
-GlobalCoinFactory.registerCoinConstructor('txlm', Txlm.createInstance);
-GlobalCoinFactory.registerCoinConstructor('dash', Dash.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tdash', Tdash.createInstance);
-GlobalCoinFactory.registerCoinConstructor('zec', Zec.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tzec', Tzec.createInstance);
-GlobalCoinFactory.registerCoinConstructor('algo', Algo.createInstance);
-GlobalCoinFactory.registerCoinConstructor('talgo', Talgo.createInstance);
-GlobalCoinFactory.registerCoinConstructor('trx', Trx.createInstance);
-GlobalCoinFactory.registerCoinConstructor('ttrx', Ttrx.createInstance);
-GlobalCoinFactory.registerCoinConstructor('xtz', Xtz.createInstance);
-GlobalCoinFactory.registerCoinConstructor('txtz', Txtz.createInstance);
-GlobalCoinFactory.registerCoinConstructor('hbar', Hbar.createInstance);
-GlobalCoinFactory.registerCoinConstructor('thbar', Thbar.createInstance);
-GlobalCoinFactory.registerCoinConstructor('ofc', Ofc.createInstance);
-GlobalCoinFactory.registerCoinConstructor('susd', Susd.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tsusd', Tsusd.createInstance);
-GlobalCoinFactory.registerCoinConstructor('cspr', Cspr.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tcspr', Tcspr.createInstance);
-GlobalCoinFactory.registerCoinConstructor('stx', Stx.createInstance);
-GlobalCoinFactory.registerCoinConstructor('tstx', Tstx.createInstance);
+// TODO: leverage statics instead of making 30 manual calls
```

Comment:
is this still a todo? if coming back to later can you add a ticket number?

Reply:
Totally, created a ticket here: https://bitgoinc.atlassian.net/browse/BG-37946

Reply:
Could you add the ticket number to the comment here?